### PR TITLE
Make aarch64 protoc compatible with RHEL 8

### PR DESCRIPTION
**Pull request by omajid**
I don't know how to trigger the build. Maybe PR CI will do it?

### CODE REVIEW

**Comment by omajid** [comment]
@tonydnewell do you know what I can do to test this locally to see if  `protoc` for arm64 is build with the correct baseline?

**Comment by tonydnewell** [comment]
To build locally you can run the following in the root of the git repository:
```bash
tools/run_tests/task_runner.py -f protoc linux aarch64
```
and the built binaries for `protoc` and the plugins will be put in the `artifacts` directory. I could do this but I can't test the images - so best if you try and build them.

Also, there is more work to do to update the docker images that are used internally - see [README.md](https://github.com/grpc/grpc/blob/master/tools/dockerfile/README.md) - however neither you or I can do this, it will need to be someone in Google:
- a version file needs to be updated: https://github.com/grpc/grpc/blob/master/tools/dockerfile/grpc_artifact_protoc_aarch64.current_version
- image needs to be added to the artifact registry

**Comment by omajid** [comment]
Thanks!

I built [tools/dockerfile/grpc_artifact_protoc_aarch64/Dockerfile](https://github.com/grpc/grpc/pull/37440/files#diff-40066e74231de485ebb577cc79b745a8d60c10cbb35a576f3e7d6f33d2cc1568) and pushed it to quay.io: https://quay.io/repository/omajid/grpc_artifact_protoc?tab=tags

I then updated current_version to use it:

```diff
--- a/tools/dockerfile/grpc_artifact_protoc_aarch64.current_version
+++ b/tools/dockerfile/grpc_artifact_protoc_aarch64.current_version
@@ -1 +1 @@
-us-docker.pkg.dev/grpc-testing/testing-images-public/grpc_artifact_protoc_aarch64:be8fc962f6a27ce976abecbda3dd02be4be5683d@sha256:a0f6b544c1b2fa75f73b26cb28281917153304e9df2789e93de0732494e6b651
\ No newline at end of file
+quay.io/omajid/grpc_artifact_protoc:latest@sha256:5d6cf2bbad8fea610909138a3e04322873aebea19761f871d858489ef6a2254a
```

I was then able to build it:

```
$ tools/run_tests/task_runner.py -f protoc linux aarch64
Will build 1 targets:
  protoc_linux_aarch64, labels ['artifact', 'protoc', 'linux', 'aarch64', 'presubmit']

START: Building targets.
PASSED: build_artifact.protoc_linux_aarch64 [time=333.4sec, retries=0:0]
SUCCESS: All targets built successfully.
```

I emulated RHEL 8 + aarch64 on my x86_64 machine using podman:

```
$ podman run --arch aarch64 -v .:/protoc/:z -it ubi8
```

And now it looks like the version of glibc is sufficient:

```
# readelf -sW protoc | grep @GLIBC | sed -E 's/.*(@GLIBC.*)/\1/' | sort -u
@GLIBC_2.17 (2)
@GLIBC_2.17 (3)
@GLIBC_2.17 (4)
# ldd protoc
        linux-vdso.so.1 (0x0000400000801000)
        librt.so.1 => /lib64/librt.so.1 (0x0000400000807000)
        libm.so.6 => /lib64/libm.so.6 (0x0000400000828000)
        libpthread.so.0 => /lib64/libpthread.so.0 (0x00004000008e9000)
        libc.so.6 => /lib64/libc.so.6 (0x000040000091e000)
        /lib/ld-linux-aarch64.so.1 (0x00007fa50b50d000)
```

So I think this automagically works.

File: tools/dockerfile/grpc_artifact_protoc_aarch64/Dockerfile [hunk 1]
```diff
@@ -13,7 +13,7 @@
 # limitations under the License.
 
 # On aarch64 the protoc and codegen plugins are built via crosscompilation.
-FROM dockcross/linux-arm64
+FROM dockcross/manylinux2014-aarch64
```
Comment (tonydnewell) [inline tools/dockerfile/grpc_artifact_protoc_aarch64/Dockerfile:16]: In the Dockerfile `grpc_artifact_python_manylinux2014_aarch64/Dockerfile` the image used is a tagged version:
`FROM dockcross/manylinux2014-aarch64:20230905-7b2d74f`
We might need to do that here to be consistent.

Comment (omajid) [reply tools/dockerfile/grpc_artifact_protoc_aarch64/Dockerfile]: Done!

**Comment by veblush** [comment]
Please resolve EasyCLA to proceed with this change. @omajid

**Comment by omajid** [comment]
@veblush done!

**Comment by yashykt** [comment]
@veblush ping

**Comment by veblush** [comment]
Sanity Checks failed. `tools/dockerfile/push_testing_images.sh` needs to be run to address this error.

**Comment by omajid** [comment]
> tools/dockerfile/push_testing_images.sh needs to be run to address this error.

Would you like me to do this? I am not sure I can push images to the registries, but I can give it a try.

**Comment by veblush** [comment]
I created https://github.com/grpc/grpc/pull/38084 for a quick merge.

**Pull request closed without merging**